### PR TITLE
[HM5] Cameras

### PR DESCRIPTION
**Pull request by mike-koch**
This should get the ball rolling regarding tracking security cameras. This should support:
 - Changing "Caught on Camera" to "Yes" when being spotted by a camera
 - Changing "Caught on Camera" to "No" when destroying a camera recorder
 - Changing "Spotted" to "Yes" if the camera detects you performing an illegal action
  - This change adds the "Witnesses" to the "Spotted" category. I don't *think* this will have any unintended consequences; however I wanted to mention it just in case.

It does seem that GitHub is having a field day with marking the entire file as different, so I'll try to see what's up with that and try to get rid of it.  Going to https://github.com/OrfeasZ/Statman/pull/10/files?utf8=✓&diff=unified&w=1 should remove the whitespace-only changes.

### CODE REVIEW

File: StatModules/HM5/Src/HM5/Hooks/Scripting.cpp [hunk 1]
```diff
@@ -1,125 +1,139 @@
...
+	else if (s_JsonEvent["Name"] == "SecuritySystemRecorder")
+	{
+		Log("ZAchievementManagerSimple Event! %s\n", s_JsonEvent.dump(4).c_str());
+		std::string s_Event = s_JsonEvent["Value"]["event"];
+
+		if (s_Event != "CameraDestroyed") 
+		{
+			std::string s_Recorded = s_Event == "spotted" ? "true" : "false";
+
+			if (s_Recorded == "true")
```
Comment (OrfeasZ) [inline StatModules/HM5/Src/HM5/Hooks/Scripting.cpp]: This here could be:
```cpp
if (s_Event == "spotted")
```

Comment (OrfeasZ) [reply StatModules/HM5/Src/HM5/Hooks/Scripting.cpp]: Also, have you checked to see if there's any events triggering after you've already destroyed the recorder and then go in the line of sight of a camera?

Comment (mike-koch) [reply StatModules/HM5/Src/HM5/Hooks/Scripting.cpp]: No events are triggered when walking into the line of sight of a camera if the recorder is already destroyed (so no need to worry about destroying the recorder, and then being "caught" on camera)

**Comment by mike-koch** [comment]
Line endings have been fixed (I ended up force pushing the branch to make sure everything got committed / pushed correctly; it shouldn't cause any issues with the merge).  I also condensed / cleaned up the one `if` condition you mentioned.

> I would suggest decoupling the witnesses event from the spotted event. If it's something you want to track it should be added separately, with a list of active witnesses maintained (similarly to how spotters are tracked).

I have separated the witnesses event from the spotted event, so they're being tracked separately now. However, I did keep the "Spotted" condition for the UI to be 0 spotters **and** 0 witnesses (to match how the game works). If you want them separate on the UI, let me know.

> As for the camera events, were you able to get the initial `SecuritySystemRecorder` `spotted` event consistently? In my testing it sometimes wouldn't trigger for the first camera in the level, even though the in-game indicator would show up. I'll also add add some more specific comments to the code.

I have not encountered any issues where the first camera does not trigger the `spotted` event yet. Is there a particular camera that seems to have this issue?

**Comment by mike-koch** [comment]
Ooops... I guess I accidentally closed this when making some branch changes on my fork 😳 ..... I'll open a new one with the latest changes.

**Pull request closed without merging**